src/ray/raylet/raylet.cc

```diff
@@ -108,6 +107,20 @@ void Raylet::Stop() {
   acceptor_.close();
 }
 
+void Raylet::TryToGetAgentInfoAndRegisterGcs() {
+  rpc::AgentInfo agent_info;
+  auto status = node_manager_.SyncGetAgentInfo(&agent_info);
+  if (status.ok()) {
+      self_node_info_.mutable_agent_info()->CopyFrom(agent_info);
+      RAY_CHECK_OK(RegisterGcs());
+  } else {
+    std::this_thread::sleep_for(std::chrono::milliseconds(20));
```

Comment:
Add some comments here about what's going on.

Reply:
Also 20ms seems too big. Maybe 1ms? Make it an os env as well.

Reply:
done.